cmd/erasure-server-pool.go

```diff
@@ -774,7 +774,13 @@ func (z *erasureServerPools) GetObjectNInfo(ctx context.Context, bucket, object
 	}
 
 	lockType = noLock // do not take locks at lower levels for GetObjectNInfo()
-	return z.serverPools[zIdx].GetObjectNInfo(ctx, bucket, object, rs, h, lockType, opts)
+	gr, err = z.serverPools[zIdx].GetObjectNInfo(ctx, bucket, object, rs, h, lockType, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	unlockOnDefer = false
+	return gr.WithCleanupFuncs(nsUnlocker), nil
```

Comment:
```suggestion
	if unlockOnDefer {
		unlockOnDefer = false
		return gr.WithCleanupFuncs(nsUnlocker), nil
	}
	return gr, nil
```

Otherwise crash.

Reply:
> Otherwise crash.
nsUnlocker is an emtpy function by default. I don't think it will crash, but yeah this is cleaner